### PR TITLE
use OrderDict because csv.DictReader returns a dict after version 3.7

### PR DESCRIPTION
**Pull request by andeman**
See: https://github.com/python/cpython/pull/8014

### CODE REVIEW

File: net/haproxy/src/opnsense/scripts/OPNsense/HAProxy/lib/haproxy/cmds.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import re
 import csv
 import json
+import collections
```
Comment (fichtner) [inline net/haproxy/src/opnsense/scripts/OPNsense/HAProxy/lib/haproxy/cmds.py:5]: Wouldn't it be cleaner to just import OrderedDict ?

Comment (fichtner) [reply net/haproxy/src/opnsense/scripts/OPNsense/HAProxy/lib/haproxy/cmds.py]: 👍

**Comment by fraenki** [comment]
LGTM, I'll apply the same fix for the statistics script.

**Comment by AdSchellevis** [comment]
I don't want to spoil the fun here, but are we sure we're fixing the right thing? The end result seems to be a dict() anyway

https://github.com/opnsense/plugins/blob/dbb697a2ec48265f47b22333fbd57afba62d5cc3/net/haproxy/src/opnsense/scripts/OPNsense/HAProxy/lib/haproxy/cmds.py#L333

**Comment by kulikov-a** [comment]
@AdSchellevis imho not just a dict() is needed, but an OrderedDict for move_to_end() to work at 
https://github.com/opnsense/plugins/blob/c149e55aa691768206a08b49229d174051b9f858/net/haproxy/src/opnsense/scripts/OPNsense/HAProxy/lib/haproxy/cmds.py#L329-L331

**Comment by AdSchellevis** [comment]
@kulikov-a it won't stick to the end when casting to dict() later on :) (if it does, it's pure luck)

**Comment by kulikov-a** [comment]
@AdSchellevis ah, got it, thanks. but shouldn't the order be preserved if (starting from 3.7) it is declared that the dict() preserves the insertion order (https://docs.python.org/3/whatsnew/3.7.html)?

**Comment by andeman** [comment]
@AdSchellevis 
This fixes the bug and don´t modifiy the existing structure as casting to a dict should preserve the order. But maybe me, @kulikov-a  and the python guys are overlooking something? Do you have any other information?

**Comment by AdSchellevis** [comment]
@kulikov-a I kind of missed that item in the release notes, my point just is that if the order doesn't matter in real life, best don't try to make it look like it does..... trusting ordering in named array types is often quite dangerous and prone to errors. Eventually it wraps into a json container, which likely doesn't care about order either.

https://github.com/opnsense/plugins/blob/c149e55aa691768206a08b49229d174051b9f858/net/haproxy/src/opnsense/scripts/OPNsense/HAProxy/lib/haproxy/cmds.py#L87

It's not my code, if @fraenki choses to accept it I don't mind, spending years on reverse engineering of all sorts of weirdness in our old code just makes my eyes hurt.

**Comment by andeman** [comment]
The code is not refactored yet. My primary goal was to get the feature to work. In the next iteration it will be cleaned.

**Comment by kulikov-a** [comment]
@AdSchellevis thanks for the explanation. thought that `json.dump()` just inherits the `dict()` behavior and now the order should be preserved for json too (https://docs.python.org/3.8/library/json.html).
@andeman sorry for the discussion. just trying to learn something

**Comment by AdSchellevis** [comment]
@kulikov-a that is good to know, but since it wasn't earlier, its likely safe to assume order doesn't matter (in which case someone could just add "id" to the container and be done with it)

**Comment by fraenki** [comment]
I can confirm that it fixes the maintenance page, so the targeted issues is solved.

**Pull request merged**